src/modules/file/file.usecase.ts

```diff
@@ -155,32 +156,77 @@ export class FileUseCases {
     );
   }
 
+  checkUpdateFileMetaDto(newFileMetadata: UpdateFileMetaDto) {
+    const metadataHasPlainName = 'plainName' in newFileMetadata;
+    const metadataHasType = 'type' in newFileMetadata;
+
+    const emptyPlainName =
+      newFileMetadata.plainName === null ||
+      newFileMetadata.plainName?.trim().length === 0;
+    const emptyType =
+      newFileMetadata.type === null ||
+      newFileMetadata.type?.trim().length === 0;
+
+    if (emptyPlainName && emptyType) {
+      throw new BadRequestException('Filename can not be empty');
+    }
+    return { metadataHasPlainName, metadataHasType, emptyPlainName, emptyType };
+  }
+
```

Comment:
Using the tools I left in other comments, I think you could get rid of this function (just take care of the type: null case sent in the body). 

Not a bad idea to re-check things in services, but If we have already class validator and some tools nestjs provides us with to check things beforehand, we can reduce code in services. More like a personal opinion tho, so take it with a grain of salt!

Reply:
yeah it sounds good to me, but i will still use the empty checks on the service just because we need to ensure that after looking into the db the update is not done with an empty filename (empty name + empty extension)